### ISSUE TITLE
Specify hashtags to be used for changesets

### ISSUE THREAD

**Issue by dev-zero** [streetcomplete/StreetComplete#2693]
It seems that for mapping events it is recommended to use a custom hashtag in the changesets.
And since StreetComplete is probably one of the easiest ways to get an average user to participate (the UX is really amazing!) it would be nice to be able to specify such a hashtag.
Maybe tying it to the session (similar to the reminder) to start, later on it might make sense to add a more elaborate interface like such that the user can set to "always use this hashtag", "use this hashtag for the next N days", "use this hashtag for this session", etc.

**Comment by westnordost** [comment]
Why can't you type the hashtag on the keyboard?

Most keyboards have the feature to add a typed word into the dictionary. So if you start typing it, it will be suggested for autocomplete. Isn't this very similar to what you are suggesting?

**Comment by matkoniecz** [comment]
> Why can't you type the hashtag on the keyboard?

I think that they want to have option to append extra text to StreetComplete changeset descriptions, right now it is not possible.

**Comment by westnordost** [comment]
Ah, I see. Well, to use hashtags for changesets in mapping events must have a reason. The reason is,to be able to ascribe all changesets that were made during a particular event to that even. But what for? I think the most important reason would be to be able to find and correct systematic mistakes made in that event, for example because the participants were taught something wrong.

This reason does not apply to StreetComplete, because the user does not map or tag anything by himself. So if anything is systematically wrong, then it is a general issue for StreetComplete, not that particular mapping event.

**Comment by dev-zero** [comment]
@westnordost since I'm new to mapping in general: what do you mean by "the user does not map or tag anything by himself"? Doesn't solving a quest lead to a changeset?

**Comment by westnordost** [comment]
Yes, but he doesn't decided what tags are used. He also does not map anything new, only add details to things that are already mapped.

**Comment by dev-zero** [comment]
I see, so hashtags are not tracked by changset (like in a commit), but rather on an object?
In that case I agree it doesn't make much sense to also add a tag with a single change of an attribute.

**Comment by matkoniecz** [comment]
hashtags are added within changeset, but it is not clear why hashtags would be so useful in StreetComplete that extra complexity would be worth implementing, maintaining and showing it to user

**Comment by dev-zero** [comment]
Hmm, but if I check [one of my changesets](https://www.openstreetmap.org/changeset/100584407) and compare with the [documentation for changesets](https://wiki.openstreetmap.org/wiki/Changeset#Tags_on_changesets) there seems to be a way to add a hashtag to a changeset.

**Comment by dev-zero** [comment]
@matkoniecz sure. Well, it is definitely not a high priority feature.

**Comment by matkoniecz** [comment]
All of tagging on StreetComplete changeset is done automatically, allowing custom additions would require extra complexity - that SC is avoiding except cases where there is strong reason to support some extra functionality.

**Comment by HolgerJeromin** [comment]
> Well, to use hashtags for changesets in mapping events must have a reason. The reason is,to be able to ascribe all changesets that were made during a particular event to that even. But what for? 

IMO the most obvious reason is to generate nice "we are the best! Look what this group has achieved in a few hours/days" statistics.

**Comment by westnordost** [comment]
Well, I guess this can be closed then. It is not really important.

**Comment by smichel17** [comment]
> It is not really important.

Not sure I agree with this. Say it turns out that in-person mapping events generate 10x more tags than online ones, or that the data quality is much better, or something like that. This is very useful to know, because it allows event organizers to focus their efforts on the most effective events. Unless there's another way to get this data, in which case I agree, it is not very important.

That said, I am skeptical of spending development time and taking up space in the main UI for a feature that I imagine is not very commonly used (…but maybe I'm wrong! I guess we couldn't tell without implementing it :wink:).

**Comment by riQQ** [comment]
StreetComplete already sets the following tags on the changesets:

tag | value
-- | --
StreetComplete:quest_type | AddWayLit
created_by | StreetComplete 31.0

**Comment by contrapunctus-1** [comment]
Speaking as someone who organizes mapping parties - some companies (which use OSM data) sponsor local mapping parties, and their only request is that communities use a specific hashtag during the events. 

The hashtag is used to measure the data contributions of events, communities, and the company's community support program.

I've used (and taught) just about every OSM editing program in existence, and have concluded that StreetComplete and MapComplete are the best options for first-time contributors. Unfortunately, neither of them supports custom changeset hashtags, which creates friction between the sponsors and the community. 

The workaround we've been using so far creates additional unpaid work for the community organizers (like me) - after each mapping party, we have to send over a list of the changesets that were made...

Hopefully this example helps the SC developers reconsider the importance of this issue.

**Comment by mnalis** [comment]
@contrapunctus-1 when new (not yet considered) evidence for some rejected feature requests appears, it is best to open new issue in StreetComplete and just reference old discussion in it, explaining why you think decision should change. (e.g. which new use case / new information / changed statistics etc). 

Commenting on closed issues/PRs is often quite likely to be overlooked.

That being said, I personally _do_ see usefulness in your suggestion and use case. But due to the way how StreetComplete implements changeset uploading (i.e. not having  simple upload dialog at the end of editing session, where such dialog could be extended like e.g. in Vespucci, iD, JOSM, ...) it would be more complicated to reasonably add it. 

Something like EveryDoor's solution (having a preference which would append such hashtag to every changeset) _might_ work, but if done that way, it is very easy to forget to remove, and keep uploading with now-incorrect changeset, which is not ideal really, especially for app specifically designed to be usable and used primarily by newbies.

In any case, StreetComplete tries _very hard_ to keep number of user settable preferences to an absolute minimum[^1], so I am not too optimistic that current decision is likely to be changed, as it would be adding a new preference for the benefit of the very few (comparatively) mappers. 

If no positive outcome comes here (i.e. issue is not reopened), I suspect you might have better luck suggesting such feature to [SCEE](https://wiki.openstreetmap.org/wiki/SCEE) -- by default it looks pretty much like StreetComplete, so it might be viable alternative if having hashtags is important to you.

[^1]: the reasoning I actually agree with, even if it often clashes with _my own_ wishes which remain unfulfilled in SC (which thus often move to SCEE)

**Comment by FloEdelmann** [comment]
Quick idea: Maybe the hashtag could (optionally) be specified when starting team mode, as an upload is (I think) automatically initiated before starting and after ending team mode.

**Comment by mnalis** [comment]
> Quick idea: Maybe the hashtag could (optionally) be specified when starting team mode, 

Nice idea, if multiple mappers are mapping in team mode, that is in fact ideal place to do it, which removes majority of the problems of people forgetting to add/remove hashtag!

Although, it might not always be the case - e.g. I've been part of mapping parties when each individual mapper got an area map with their own region which they should survey. I dunno how prevalent that is  compared to multiple-people-mapping-same-area. I don't know which model @contrapunctus-1 mapping parties follow in particular.

> as an upload is (I think) automatically initiated before starting and after ending team mode.

I don't think so -- upload of solved quests does not seem to be automatically initiated for me in SC v62.1 (although there is some network activity upon exiting the team mode (i.e. star-spinner rotates), my unuploaded quests still remain unuploaded. But that behaviour could potentially be changed if hashtag was entered... (Assuming vast majority of such organized mapping always have a team of people going together to same places)

**Comment by contrapunctus-1** [comment]
Thanks for your thoughtful responses, @mnalis . 

We indeed avoid the use of Team Mode, as we split participants up into teams, with each team having users of different editors. That way, each newcomer is able to enjoy the full variety of quests offered by SC out of the box.

Re: forgetting to turn off the hashtags...

1. Every Door has an option to confirm the use of the hashtags before each changeset creation. Combining that with SC's one-changeset-per-quest-*type* behavior will prompt the user at seemingly unpredictable times, but could still work.

2. Or, when hashtags are enabled, SC could display a "Your changes will be uploaded with \<hashtags>" message at the edge of the screen, which the user could tap to go to the relevant spot in the settings menu.

**Comment by riQQ** [comment]
Another option would be, when auto-upload is turned off, to (optionally?) show a dialog to specify a changeset hashtag when uploading the changes.

**Comment by mnalis** [comment]
> Another option would be, when auto-upload is turned off, to (optionally?) show a dialog to specify a changeset hashtag when uploading the changes.

Maybe, but:
- doing it _**unconditionally**_ in StreetComplete (i.e. adding an extra step for 99% of the users for benefit of 1% of the users) would likely be considered a bad trade
- adding a preference to do it only for users who _**opt-in**_ would solve that problem, but StreetComplete is usually averse to adding any kind of preferences (and _especially_ when such preference would only going to benefit small percentage of mappers)
  Such a preference might be an option for SCEE, though; so if such solution would work for @contrapunctus-1, they could open an issue there to request that feature.

**Comment by sfkeller** [comment]
My main argument in favor of an opt-in setting of a hashtag is, that if this feature is missing, StreetComplete is not eligible for "[directed editing](https://wiki.osmfoundation.org/wiki/Directed_Editing_Policy)" like Mapping Parties or Mapathons, since these action require to set a hashtag.

**Comment by westnordost** [comment]
I don't think direction is necessary when using StreetComplete, as users don't need to learn to use the editor and tag things correctly. They just need to answer the questions asked to their best knowledge.

**Comment by sfkeller** [comment]
That's not the point. The point is that "changeset comments should include the unique hashtag" (see "Organised Editing Guidelines"). In other words, StreetComplete isn't really suitable for mapping parties or mapathons without the ability to use hashtags, which is actually too bad.

**Comment by westnordost** [comment]
Well, I don't think it applies. And if it applies, it shouldn't apply, because there is noone that directs users *how* to tag things and for me, that would be the important bit that necessitates a directed editing policy. StreetComplete users *can't* add any novel tags, or map in a way that would not be accepted by the OSM community. It matters little whether they are directed to only map street surfaces or to only map baby changing tables.

@matkoniecz told me that maybe he will offer a workshop at the SotM to take a mapping-walk with StreetComplete. Is that "directed editing" for you that would require the bureaucracy associated with directed editing? If yes, then I'd say, the OSMF successfully shot itself into its foot with that policy.

**Edit**: Also, I notice that the policy you linked is deprecated. <i>"This document was proposed but has not become an OSMF policy."</i>

**Comment by matkoniecz** [comment]
https://osmfoundation.org/wiki/Organised_Editing_Guidelines is the active one with not really clear definition from what I see (sadly - I faintly remember complaining about it when it was proposed)

> They are not meant to apply to community activities like mapping parties between friends or doing a presentation on OSM at a local club.

**Comment by sfkeller** [comment]
> Well, I don't think it applies. And if it applies, it shouldn't apply, because there is noone that directs users how to tag things and for me

Here's another attempt to explain the issue: After a mapping party or a mapathon, how would you overview, manage, analyze and share the results and work done of the event? Here's how we do it (= using a changeset hashtag): https://resultmaps.neis-one.org/osm-changesets?comment=MapathonRappi

**Comment by westnordost** [comment]
If you are after statistics, couldn't you just filter by 1. the usernames of people that attended and 2. the time frame in which the mapathon etc. was held (and 3. by user agent "StreetComplete")?

**Comment by sfkeller** [comment]
You mean that instead of a single call with a hashtag (as described above, without any coding), every organizer who uses StreetComplete in a Mapathon or Mapping Party should write a script where they have to specify all participant names individually Sure, that’s possible. But some people might not be able to do that.  And as I said: Unfortunately, StreetComplete cannot be used in organized editing actions because it does not comply with the "Organized Editing Guidelines", which require editors to use “a unique hashtag in the changeset comments.”

My suggestion really is, be able to change the default changeset comment of StreetComplete. This would benefit not only mapathons and mapping parties, but also many other initiatives using StreetComplete.
